src/components/navBar/index.tsx

```diff
@@ -1,17 +1,94 @@
-import Image from 'next/image';
-import styles from '@/components/navBar/navBar.module.scss';
+import { useEffect, useState } from "react";
+import Image from "next/image";
+import userData from "@/helperFunctions/getUser";
+import { toast, ToastTypes } from "@/helperFunctions/toast";
+import navLinks from "./navLinks";
+import NavbarLinks from "./NavbarLinks";
+import UserSignIn from "./UserSignIn";
+import {
+  RDS_LOGO,
+  HAMBURGER_ICON,
+  SIGN_IN_WITH_GITHUB,
+  SIGN_IN,
+  HOME_PAGE_LINK,
+  TWENTY_FIVE_PX,
+  THIRTY_PX,
+  FIFTY_PX,
+} from "@/components/constants/navbar";
+import styles from "@/components/navBar/navBar.module.scss";
 
-const RDSLogo = '/RDSLogo.png';
+const NavBar = () => {
+  const [toggle, setToggle] = useState(false);
+  const [userIsLoggedIn, setUserIsLoggedIn] = useState(false);
+  const [userDetails, setUserDetails] = useState({
+    userName: "",
+    firstName: "",
+    profilePicture: "",
+  });
 
-const NavBar = () => (
-  <nav className={styles.navBar}>
-    <a className={styles.logo} href="https://realdevsquad.com"><Image width="45px" height="45px" src={RDSLogo} alt="real-dev squad" /></a>
-    <a href="https://welcome.realdevsquad.com/">Welcome</a>
-    <a href="https://www.realdevsquad.com/events.html">Events</a>
-    <a href="https://members.realdevsquad.com/">Members</a>
-    <a href="https://crypto.realdevsquad.com/">Crypto</a>
-    <a className={styles.active} href="https://status.realdevsquad.com/">Status</a>
-  </nav>
-);
+  useEffect(() => {
+    (async () => {
+      await userData()
+        .then((data) => {
+          const { username, first_name, picture } = data;
+          setUserDetails({
+            userName: username,
+            firstName: first_name,
+            profilePicture: picture?.url,
+          });
+
+          setUserIsLoggedIn(true);
+        })
+        .catch((error) => {
+          toast(ToastTypes.ERROR, (error as Error).message);
+        });
+    })();
+  }, [userDetails, userIsLoggedIn]);
+
+  return (
+    <nav className={styles.navBar}>
+      <button className={styles.hamburger} onClick={() => setToggle(!toggle)}>
+        <Image
+          src={HAMBURGER_ICON}
+          alt="Hamburger"
+          height={THIRTY_PX}
+          width={TWENTY_FIVE_PX}
+        />
+      </button>
+
+      <ul
+        className={
+          toggle ? `${styles.navBarMenu} ${styles.active}` : styles.navBarMenu
+        }
+      >
+        <li className={styles.navBarLogoLi}>
+          <a href={HOME_PAGE_LINK}>
+            <Image
+              src={RDS_LOGO}
+              alt="Real_Dev_Squad"
+              height={FIFTY_PX}
+              width={FIFTY_PX}
+            />
+          </a>
+        </li>
+        <NavbarLinks data={navLinks} />
+        <li className={styles.navBarLoginLi}>
```

Comment:
This should be a separate component - GitHub Sign or SignUp.

Reply:
This is just a wrapper around a reusable component to set the `display: none` property according to screen sizes. I guess more nesting is not required here. Also why it is `li` not `div` or any other semantic tag, because the `NavbarLinks` component consists of `li` tag elements and all this is wrapped around `ul`